react-validated-input-component/src/validated-input.jsx

```diff
@@ -0,0 +1,56 @@
+import React, { useState } from 'react';
+
+export function ValidatedInput() {
+  const [pw, setPw] = useState('');
+
+  function handlePasswordChange(event) {
+    setPw(event.target.value);
+  }
+
+  function textCounter() {
```

Comment:
These checks are good, but they repeat the validation code. You might consider having all the validation code in one function that calculations both the icon classname and the display text.